proctor-common/src/main/java/com/indeed/proctor/common/ForceGroupsOptionsStrings.java

```diff
@@ -17,60 +23,206 @@
+        int bucketValueStart = groupString.length() - 1;
+        for (; bucketValueStart >= 0; bucketValueStart--) {
+            if (!Character.isDigit(groupString.charAt(bucketValueStart))) {
+                break;
             }
-            //  minus sign can only be at the beginning of a run
-            if (piece.charAt(bucketValueStart) != '-') {
-                bucketValueStart++;
+        }
+        //  if no name or no value was found, it's not a valid proctor test bucket name
+        if ((bucketValueStart == groupString.length() - 1) || (bucketValueStart < 1)) {
+            return Optional.empty();
+        }
+        //  minus sign can only be at the beginning of a run
+        if (groupString.charAt(bucketValueStart) != '-') {
+            bucketValueStart++;
+        }
+
+        return Optional.of(bucketValueStart);
+    }
+
+    /**
+     * The format of the payloadString is the following:
+     * <ul>
+     *     <li>payloadType:payloadValue</li>
+     *     <li>Where payloadType is one of the 7 payload types supported by Proctor (stringValue, stringArray, doubleValue, longValue, longArray, map)</li>
+     *     <li>If payloadValue is an array is expected in the following format: [value,value,value]</li>
+     *     <li>If payloadValue is a map expecting following format: [key:value,key:value,key:value]</li>
+     * </ul>
+     */
+    @Nullable
+    public static Payload parseForcePayloadString(final String payloadString )
+    {
+        final Payload payload = new Payload();
+        final String[] payloadPieces = payloadString.split(":",2);
+        try {
+            final PayloadType payloadType = PayloadType.payloadTypeForName(payloadPieces[0]);
+            final String payloadValue = payloadPieces[1];
+            switch (payloadType) {
+                case DOUBLE_VALUE:
+                {
+                    payload.setDoubleValue(Double.parseDouble(payloadValue));
+                    break;
+                }
+                case DOUBLE_ARRAY:
+                {
+                    payload.setDoubleArray(Arrays.stream(getPayloadArray(payloadValue))
+                            .map(Double::valueOf)
+                            .toArray(Double[]::new) );
+                    break;
+                }
+                case LONG_VALUE:
+                {
+                    payload.setLongValue(Long.parseLong(payloadValue));
+                    break;
+                }
+                case LONG_ARRAY:
+                {
+                    payload.setLongArray(Arrays.stream(getPayloadArray(payloadValue))
+                            .map(Long::valueOf)
+                            .toArray(Long[]::new));
+                    break;
+                }
+                case STRING_VALUE:
+                {
+                    // Remove quotes
+                    payload.setStringValue(payloadValue.substring(1,payloadValue.length()-1));
```

Comment:
What will happen if `stringValue:x` (single character string without quotes) is an input of this method? Will this be caught with `catch(...)` at line 186?

Reply:
Considering this more, I feel like `ObjectMapper` is a good option here (for all types except Map) instead of parsing by ourselves because all values are valid as JSON.

maybe something like
```
case DOUBLE_VALUE:
  payload.setDoubleValue(mapper.readValue(payloadValue, Double.class));
case DOUBLE_ARRAY:
  payload.setDoubleArray(mapper.readValue(payloadValue, Double[].class));
...(same for Long and String)
```

In this way, we can remove two methods `getPayloadArray` and `getPayloadStringArray` from this class.

Reply:
This will make some behavior difference though. For example, an input `stringValue:"foo\"bar"` will be parsed as `foo"bar` instead of `foo\"bar`, but I think the former behavior is more intuitive.

Reply:
Just to confirm:
-  `String test = "stringValue:\"foo\"bar\"";` would be parsed as Java string `"foo"`. 
-  `String test = "stringValue:\"foo\\\"bar\"";` would be parsed as Java string`"foo\"bar"`